src/firefly/java/edu/caltech/ipac/util/FormatUtil.java

```diff
@@ -0,0 +1,224 @@
+package edu.caltech.ipac.util;
+
+import edu.caltech.ipac.firefly.server.db.DuckDbAdapter;
+import edu.caltech.ipac.firefly.server.db.DuckDbReadable;
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.table.DataGroup;
+
+import javax.annotation.Nonnull;
+import java.io.BufferedReader;
+import java.io.CharArrayReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+import static edu.caltech.ipac.table.IpacTableUtil.isIpacTable;
+import static edu.caltech.ipac.util.FormatUtil.Format.*;
+
+/**
+ * A collection of utilities related to File Format
+ *
+ * @author loi
+ * @version : $
+ */
+public class FormatUtil {
+    private static final int SAMPLE_SIZE = (int) (8 * FileUtil.K);
+    private static final Logger.LoggerImpl LOGGER = Logger.getLogger();
+
+    public enum Format {
+        TSV("tsv", ".tsv"),
+        CSV("csv", ".csv"),
+        IPACTABLE("ipac", ".tbl"),
+        UNKNOWN("null", null),
+        FIXEDTARGETS("fixed-targets", ".tbl"),
+        FITS("fits",".fits"),
+        JSON("json", ".json"),
+        PDF("pdf", ".pdf"),
+        TAR("tar", ".tar"),
+        JAR("jar", ".jar"),
+        HTML("html", ".html"),
+        VO_TABLE("votable", ".xml"),
+        VO_TABLE_TABLEDATA("votable-tabledata", ".vot"),
+        VO_TABLE_BINARY("votable-binary-inline", ".vot"),
+        VO_TABLE_BINARY2("votable-binary2-inline", ".vot"),
+        VO_TABLE_FITS("votable-fits-inline",".vot"),
+        REGION("reg", ".reg"),
+        PNG("png", ".png"),
+        JPEG("jpeg", ".jpg"),
+        UWS("uws", ".xml"),
+        PARQUET(DuckDbReadable.Parquet.NAME, "."+DuckDbReadable.Parquet.NAME),
+        ZIP("zip", ".zip"),
+        GZIP("gzip", ".gz");
+
+        public final String type;
+        final String fileNameExt;
+        Format(String type, String ext) {
+            this.type = type;
+            this.fileNameExt = ext;
+        }
+        public String getFileNameExt() {
+            return fileNameExt;
+        }
+        public String toString() {return type;}
+    }
+
+    /**
+     * Determines the MIME type of the given file.
+     *
+     * @param inFile input file to detect
+     * @return A String representing the MIME type of the file, or "application/x-unknown" otherwise
+     */
+    public static DuckDbAdapter.MimeDesc getMimeType(File inFile) {
+        return DuckDbAdapter.getMimeType(inFile);
+    }
+
+    /**
+     * Detects the format of the given file
+     *
+     * @param inFile The full path of the file to detect the format for.
+     * @return The detected {@code Format} of the file, or {@code UNKNOWN} if the format could not be determined.
+     * @throws IOException If an I/O error occurs while accessing the file.
+     */
+    @Nonnull
+    public static Format detect(File inFile) throws IOException {
+
+        Format format = null;
+        DuckDbAdapter.MimeDesc mimeDesc = DuckDbAdapter.getMimeType(inFile);
```

Comment:
So this line actually looks inside the file? The function name is a little confusing since it implies it is just looking at the extension.

Reply:
This line uses DuckDB's magic extension to determine the file's MIME type, similar to the Unix `file` command. So yes, it will read the magic number(if exists) along with additional data as needed.